Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm

```diff
@@ -203,7 +204,7 @@
     else if (webAudioCount)
         category = AudioSession::CategoryType::AmbientSound;
 
-    if (category == AudioSession::CategoryType::None && m_previousCategory != AudioSession::CategoryType::None) {
+    if (category == AudioSession::CategoryType::None && m_previousCategory != AudioSession::CategoryType::None && m_previousCategory != AudioSession::CategoryType::PlayAndRecord) {
```

Comment:
Why do we not arm the timer when changing from `PlayAndRecord` to `None`?

Reply:
The reason is that, when setMicrophoneActive(false) resolves, we want to be sure to move out of PlayAndRecord.
This allows web applications that want to capture/record microphone then play the recorded audio, to play in regular MediaPlayback category.
With the timer approach, this would be hit and miss (or we would need to add logic to resolve the promise once the timer fires).